### PR TITLE
Feat: Add a `yarn dev` cmd for local dev without Docker

### PR DESCRIPTION
**Pull request by rsylvian**
# Description
- Add a `yarn dev` command that runs everything needed locally
- Add documentation to `local-development.md`

# Reasons
It seems that Jekyll & Docker don't mix well together and so this PR creates a new command for those that wish to run everything locally. The first results are pretty exciting because caching is done locally and build tools can use the full power of the machine (instead of only half allocated CPUs in Docker). The very first generation is taking the usual duration (120-140s) but the following ones are blazing fast. It seems like content changes are `under <5s`, CSS changes `under <1s` and js changes around `2-3s`.

Special thanks to @devinmarieb, @chantelle11, @teesloane & @tshen815 for beta-testing the changes and helping me uncovering some issues ahead of time 🙌

### CODE REVIEW

File: jekyll/_config.yml [hunk 1]
```diff
@@ -155,6 +155,12 @@ defaults:
     values:
       suggested: false
 
+  # Disable sitemap by default (only want in prod)
+  - scope:
+      path: ""
+    values:
+      sitemap: false
+
```
Comment (tshen815) [inline jekyll/_config.yml:163]: Just curious, why do we want to disable sitemaps here only to enable it in `_config_production.yml` for prod?

Comment (rsylvian) [reply jekyll/_config.yml:163]: Generating the sitemap in local takes between 12 and 15 seconds for **each** rebuild. We only need the sitemap when we are building prod so this is why I disabled it

File: jekyll/_config.yml [hunk 2]
```diff
@@ -170,3 +176,5 @@ rollbar_env: development
 optimizely: "GfYszBpMkefBSwiiEkH3b3"
 
 # server_version: "2.18.4"
+
+sitemap: false
```
Comment (rsylvian) [inline jekyll/_config.yml:180]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -13,12 +13,15 @@
     "datadog-synthetics-ci": "yarn datadog-ci synthetics run-tests -f ./.datadog/development.synthetics.json --tunnel",
     "datadog-synthetics-preview": "node ./scripts/datadog-synthetics-preview.js && yarn datadog-ci synthetics run-tests -f ./.datadog/preview.synthetics.json",
     "datadog-synthetics-prod": "yarn datadog-ci synthetics run-tests -f ./.datadog/production.synthetics.json",
-    "dev": "yarn webpack-dev && docker-compose up",
+    "caddy-dev": "caddy run --config ./Caddyfile-dev",
+    "jekyll-dev": "bundle exec jekyll serve -s jekyll --incremental --host=0.0.0.0",
+    "webpack-dev": "webpack --mode=development",
```
Comment (ccgeller) [inline package.json:18]: From @teesloane "webpack-dev": "webpack --watch --mode=development",

**Review by tshen815 — APPROVED** [review]
89 seconds to start and 2.5 seconds for a change in `index.html` to regenerate! THanks for working on this, this is amazing! 🎉 🤩

File: docs/local-development.md [hunk 4]
```diff
@@ -23,15 +23,24 @@ _(Warning: This may take up to 10 minutes to build)_
 
 **Note:** In the event you find yourself needing to cleanup docker/jekyll cache, you can use the `yarn clean` command.
 
-## 2. Local Development with Ruby and Bundler (alternative to Docker)
+## 2. Local Development with Ruby and Bundler
 
-If you already have a stable Ruby environment (currently Ruby 2.7.4) and feel comfortable installing dependencies, install Jekyll by following [this guide](https://jekyllrb.com/docs/installation/).
+If you already have a stable Ruby environment (currently Ruby 2.7.4) and feel comfortable installing external dependencies, you can follow these instructions. Check out the [Gemfile](https://github.com/circleci/circleci-docs/blob/master/Gemfile) for the Ruby version we're currently using. We recommend [RVM](https://rvm.io/) for managing multiple Ruby versions.
```
Comment (thomasr8) [inline docs/local-development.md:28]:
```suggestion
If you already have a stable Ruby environment (currently Ruby 2.7.4) and feel comfortable installing external dependencies, you can follow these instructions. Check out the [Gemfile](https://github.com/circleci/circleci-docs/blob/master/Gemfile) for the Ruby version we are currently using. We recommend [RVM](https://rvm.io/) for managing multiple Ruby versions.
```

File: docs/local-development.md [hunk 5]
```diff
@@ -23,15 +23,24 @@ _(Warning: This may take up to 10 minutes to build)_
...
+2. Clone the CircleCI docs repo: `git clone https://github.com/circleci/circleci-docs.git`
+3. Start Docker Desktop
+4. Add the following line to your `/etc/hosts` file:
+   ```bash
+   127.0.0.1 ui.circleci.com
+   ```
+5. Run `bundle install`. You will need to have [Bundler](https://bundler.io/) installed on your local machine.
+6. Run `yarn install`. You will need to have [Yarn](https://yarnpkg.com/getting-started) installed on your local machine.
+7. Finally, run `yarn dev` to start the site.
+8. The docs site will now be running on <https://ui.circleci.com/docs/>. If the browser presents to you an HSTS Security Warning, you can safely bypass it as it is an expected outcome of running the Caddy Reverse Proxy in Docker.
```
Comment (thomasr8) [inline docs/local-development.md:40]:
```suggestion
8. The docs site will now be running on <https://ui.circleci.com/docs/>. If the browser displays a HSTS Security Warning, you can safely bypass it, as it is an expected outcome of running the Caddy Reverse Proxy in Docker.
```

**Review by thomasr8 — APPROVED** [review]
Just a couple of minor suggestions. This looks really good 👍

**Pull request merged**